### PR TITLE
Add the backoffice DS ballerina implementations

### PR DESCRIPTION
**Pull request by 1akshitha**
## Purpose
This PR adds the bindgen files and backoffice ballerina service. 
Also it fixes some NPEs found in the java layer.

### CODE REVIEW

File: backoffice/backoffice-domain-service/ballerina/modules/backoffice/Module.md [hunk 1]
```diff
@@ -0,0 +1,6 @@
+Prints "Hello, World!" with a main function.
```
Comment (tharindu1st) [inline backoffice/backoffice-domain-service/ballerina/modules/backoffice/Module.md:1]: Lets have proper md file

File: backoffice/backoffice-domain-service/ballerina/modules/backoffice/backoffice-api_service.bal [hunk 2]
```diff
@@ -0,0 +1,117 @@
+import ballerina/http;
```
Comment (tharindu1st) [inline backoffice/backoffice-domain-service/ballerina/modules/backoffice/backoffice-api_service.bal:1]: license header

File: backoffice/backoffice-domain-service/ballerina/modules/backoffice/backoffice-api_service.bal [hunk 3]
```diff
@@ -0,0 +1,117 @@
...
+    cors: {
+        allowOrigins: ["*"],
+        allowCredentials: true,
+        allowHeaders: ["*"],
+        exposeHeaders: ["*"],
+        maxAge: 84900
+    }
+}
+
+service /api/am/backoffice/v1 on ep0 {
```
Comment (tharindu1st) [inline backoffice/backoffice-domain-service/ballerina/modules/backoffice/backoffice-api_service.bal:21]: we can remove v1

File: backoffice/backoffice-domain-service/ballerina/modules/backoffice/init.bal [hunk 4]
```diff
@@ -0,0 +1,37 @@
...
+configurable DatasourceConfiguration datasourceConfiguration = ?;
+
+function init() {
+    io:println("Starting APK Backoffice Domain Service...");
+    APKConfiguration apkConfig = {
+        datasourceConfiguration: datasourceConfiguration
+    };
+    string configJson = apkConfig.toJson().toJsonString();
+    // Pass the configurations to java init component
+    io:println(configJson);
```
Comment (tharindu1st) [inline backoffice/backoffice-domain-service/ballerina/modules/backoffice/init.bal:32]: lets remove this

File: backoffice/backoffice-domain-service/ballerina/modules/backoffice/init.bal [hunk 5]
```diff
@@ -0,0 +1,37 @@
...
+    api:APIManagementException? err = apkinit:APKComponent_activate(configJson);
+    if (err != ()) {
+        io:println(err);
```
Comment (tharindu1st) [inline backoffice/backoffice-domain-service/ballerina/modules/backoffice/init.bal:35]: use log library

File: common-java-libs/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/APIManagerFactory.java [hunk 6]
```diff
@@ -108,10 +108,31 @@ public APIConsumer getAPIConsumerFromCache(String key, ConsumerCreator consumerC
         return consumer;
     }
 
+    public APIProvider getAPIProviderFromCache(String key, ProviderCreator providerCreator)
+            throws APIManagementException {
+        APIProvider provider = providers.get(key);
```
Comment (tharindu1st) [inline common-java-libs/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/APIManagerFactory.java:113]: lets remove this cache

**Review by tharindu1st — CHANGES_REQUESTED** [review]

File: backoffice/backoffice-domain-service/ballerina/modules/backoffice/backoffice-api_service.bal (hunk 2 above)
Comment (1akshitha) [reply backoffice/backoffice-domain-service/ballerina/modules/backoffice/backoffice-api_service.bal:19]: added with f0cddb250859adb268acb465a0f427f1da0e5766

File: backoffice/backoffice-domain-service/ballerina/modules/backoffice/Module.md (hunk 1 above)
Comment (1akshitha) [reply backoffice/backoffice-domain-service/ballerina/modules/backoffice/Module.md]: Removed with f0cddb250859adb268acb465a0f427f1da0e5766, until a proper md added

File: backoffice/backoffice-domain-service/ballerina/modules/backoffice/backoffice-api_service.bal (hunk 3 above)
Comment (1akshitha) [reply backoffice/backoffice-domain-service/ballerina/modules/backoffice/backoffice-api_service.bal]: addressed with f0cddb250859adb268acb465a0f427f1da0e5766

File: backoffice/backoffice-domain-service/ballerina/modules/backoffice/init.bal (hunk 4 above)
Comment (1akshitha) [reply backoffice/backoffice-domain-service/ballerina/modules/backoffice/init.bal]: addressed with f0cddb250859adb268acb465a0f427f1da0e5766

File: backoffice/backoffice-domain-service/ballerina/modules/backoffice/init.bal (hunk 5 above)
Comment (1akshitha) [reply backoffice/backoffice-domain-service/ballerina/modules/backoffice/init.bal]: addressed with f0cddb250859adb268acb465a0f427f1da0e5766

File: common-java-libs/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/APIManagerFactory.java (hunk 6 above)
Comment (1akshitha) [reply common-java-libs/org.wso2.apk.apimgt.impl/src/main/java/org/wso2/apk/apimgt/impl/APIManagerFactory.java]: addressed with f0cddb250859adb268acb465a0f427f1da0e5766

**Pull request merged**